### DISCUSSION TITLE
npx create-next-app command not working

### DISCUSSION THREAD

**Discussion by kunaaal13** [vercel/next.js#35611] [Help] [answered]
This is showing this error

> Creating a new Next.js app in /Users/kunalyadav/Desktop/App/bye.
> 
<img width="986" alt="Screenshot 2022-03-26 at 12 13 14 PM" src="https://user-images.githubusercontent.com/84015751/160228191-e7ccd74a-0797-4b95-b710-a47d0b97a2b4.png">

> Using yarn.
> 
> Installing dependencies:
> - react
> - react-dom
> - next
> 
> node:events:504
>       throw er; // Unhandled 'error' event
>       ^
> 
> Error: spawn yarn ENOENT
>     at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
>     at onErrorNT (node:internal/child_process:478:16)
>     at processTicksAndRejections (node:internal/process/task_queues:83:21)
> Emitted 'error' event on ChildProcess instance at:
>     at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
>     at onErrorNT (node:internal/child_process:478:16)
>     at processTicksAndRejections (node:internal/process/task_queues:83:21) {
>   errno: -2,
>   code: 'ENOENT',
>   syscall: 'spawn yarn',
>   path: 'yarn',
>   spawnargs: [
>     'add',
>     '--exact',
>     '--cwd',
>     '/Users/kunalyadav/Desktop/App/bye',
>     'react',
>     'react-dom',
>     'next'
>   ]
> }

**Comment by Mantofka** [comment]
Hey, try to write `npx create-next-app --use-npm` after you press enter, you will be prompted to write down the title. Hope it works

**Reply by danielru15** [reply]
sabe otra solucuion que no implique el --use-npm? si uso yarn

**Comment by ijjk** [comment, marked as answer]
Hi, this has been fixed in the latest version of Next.js `v12.1.2` please give it a try! Note: this may require running with `@latest` e.g. `npx create-next-app@latest`. Specific PR this was fixed in can be seen here https://github.com/vercel/next.js/pull/35608

**Reply by Pratap-Jitendra** [reply]
Dear all,

plese resolve my issue for react js on npm run deploy

node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: spawn git ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn git',
  path: 'git',
  spawnargs: [ 'config', 'user.name' ]
}

**Reply by Pratap-Jitendra** [reply]
on deploying time after pushing well but publish even link generated but only read only message comes.
 sample@0.1.0 deploy
> gh-pages -d build

node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: spawn git ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn git',
  path: 'git',
  spawnargs: [ 'config', 'user.name' ]
}

**Reply by Pratap-Jitendra** [reply]
I am suffering about 1 months please solve my issue anyone.
C:\Users\hp\Desktop\firstsample\sample\package.json

**Reply by Pratap-Jitendra** [reply]
{
  "homepage": "https://Pratap-Jitendra.github.io/sample",
  "name": "sample",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "gh-pages": "^4.0.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
    "web-vitals": "^2.1.4"
  },
  "scripts": {
    "predeploy": "npm run build",
    "deploy": "gh-pages -d build",
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
